store/README.md

```diff
@@ -0,0 +1,167 @@
+But if you intend on running Store as a separated process, you need to install and run it separately:
+
+```sh
+# Installs `miden-node-store` executable
+cargo install --path store
+```
+
+In order to run Store, you must provide genesis file. To generate genesis file you need to use [Miden node](../README.md#generating-the-genesis-file)'s `make-genesis` command. 
+
+
+You'll also need to provide a configuration file. We have an example config file in [store-example.toml](store-example.toml).
+
+Then, to run the Store:
+
+```sh
+miden-node-store serve --config <path-to-store-config-file>
+```
+
+## API
+
+**Store** serves connections using [gRPC protocol](https://grpc.io) on a port, set in configuration file. Here is a brief
+description of supported methods.
+
+### ApplyBlock
+
+Applies changes of a new block to the DB and in-memory data structures.
+
+**Parameters**
+
+* `block`: `BlockHeader` – block header ([src](../proto/proto/block_header.proto)).
+* `accounts`: `[AccountUpdate]` – a list of account updates.
+* `nullifiers`: `[Digest]` – a list of nullifier hashes.
+* `notes`: `[NoteCreated]` – a list of notes created.
+
+**Returns**
+
+This method doesn't return any data.
+
+### CheckNullifiers
+
+Get a list of proofs for given nullifier hashes, each proof as Tiered Sparse Merkle Trees ([read more](../proto/proto/tsmt.proto)).
+
+**Parameters:**
+
+* `nullifiers`: `[Digest]` – array of nullifier hashes.
+
+**Returns:**
+
+* `proofs`: `[NullifierProof]` – array of nullifier proofs, positions correspond to the ones in request.
+
+### GetBlockHeaderByNumber
+
+Retrieves block header by given block number.
+
+**Parameters**
+
+* `block_num`: `uint32` *(optional)* – the block number of the target block. If not provided, the latest known block will be returned.
+
+**Returns:**
+
+* `block_header`: `BlockHeader` – block header.
+
+### GetBlockInputs
+
+Returns data needed by the block producer to construct and prove the next block.
+
+**Parameters**
+
+* `account_ids`: `[AccountId]` – array of account IDs. 
+* `nullifiers`: `[Digest]` – array of nullifier hashes (not currently in use).
+
+**Returns**
+
+* `block_header`: `[BlockHeader]` – the latest block header.
+* `mmr_peaks`: `[Digest]` – peaks of the above block's mmr, The `forest` value is equal to the block number.
+* `account_states`: `[AccountBlockInputRecord]` – the hashes of the requested accouts and their authentication paths.
+* `nullifiers`: `[NullifierBlockInputRecord]` – the requested nullifiers and their authentication paths.
+
+### GetTransactionInputs
+
+Returns account and nullifiers descriptors. 
+
+**Parameters**
+
+* `account_id`: `AccountId` – account ID.
+* `nullifiers`: `[Digest]` – array of nullifier hashes.
+
+**Returns**
+
+* `account_state`: `AccountTransactionInputRecord` – account's descriptors. 
+* `nullifiers`: `[NullifierTransactionInputRecord]` – the requested nullifiers' blocks at which ones have been consumed, zero if not consumed.
```

Comment:
Could we double-check if the description here is accurate? I think the nullifier record might return the block number at which the nullifier was recorded (rather than 0 or 1).

Reply:
@bobbinth yes, you're right, I will rephrase it in order to be less confusing.